File: app/src/main/java/com/github/weatherapp/data/SharedPrefsAppSettings.java
package com.github.weatherapp.data;

import android.content.SharedPreferences;

import com.github.weatherapp.core.AppSettings;

public class SharedPrefsAppSettings implements AppSettings {
    private static final String KEY_METRIC = "PREFERENCE_METRIC";
    private final SharedPreferences preferences;

    public SharedPrefsAppSettings(SharedPreferences preferences) {
        this.preferences = preferences;
    }

    @Override
    public String getMetricUnit() {
        return preferences.getString(KEY_METRIC, "metric");
    }

    @Override
    public void setMetricUnit(String unit) {
        preferences.edit()
                .putString(KEY_METRIC, unit)
                .apply();
    }
}

File: app-kotlin/src/main/kotlin/com/github/weatherapp/data/SharedPrefsAppSettings.kt
package com.github.weatherapp.data

import android.content.SharedPreferences

import com.github.weatherapp.core.AppSettings

class SharedPrefsAppSettings(private val preferences: SharedPreferences) : AppSettings {

    override var metricUnit: String
        get() = preferences.getString(KEY_METRIC, "metric")
        set(unit) = preferences.edit().putString(KEY_METRIC, unit).apply()

    companion object {
        private val KEY_METRIC = "PREFERENCE_METRIC"
    }
}
